compiler/p/codegen/PPCBinaryEncoding.cpp

```diff
@@ -1019,6 +1019,20 @@ static void fillFieldR(TR::Instruction *instr, uint32_t *cursor, uint32_t val)
    cursor[0] |= (val << 20);
    }
 
+/**
+ * Fills in the 8-bit IMM8 field of a binary-encoded instruction with the provided immediate value:
+ *
+ * +----------------------+---------------------------------------+
+ * |                      | IMM8            |                     |
+ * | 0                    | 13              | 21                  |
+ * +----------------------+---------------------------------------+
+ */
+static void fillFieldIMM8(TR::Instruction *instr, uint32_t *cursor, uint32_t val)
+   {
+   TR_ASSERT_FATAL_WITH_INSTRUCTION(instr, (val & 0xffu) == val, "0x%x is out-of-range for IMM8(8) field", val);
```

Comment:
I see there was some discussion previously about accepting both signed and unsigned immediates here (and the encoder tests seem to indicate that this field _should_ accept signed immediates), but this check will only allow 8 bit unsigned immediates.